orttraining/orttraining/training_ops/cuda/activation/bias_gelu_grad_impl.cu

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "orttraining/training_ops/cuda/activation/bias_gelu_grad_impl.h"
+
+#include "core/providers/cuda/cu_inc/common.cuh"
+#include "core/providers/cuda/shared_inc/fast_divmod.h"
+#include "orttraining/training_ops/cpu/activation/gelu_computation_mode.h"
+#include "orttraining/training_ops/cuda/activation/gelu_grad_impl_common.cuh"
+
+namespace onnxruntime {
+namespace cuda {
+
+template <typename T, typename GeluComputationMode, int num_consecutive_elements_per_group, int num_groups_per_thread>
```

Comment:
the element-wise iteration logic is a bit tricky. i had to fix several issues with it. please take a closer look when reviewing